signac/contrib/filterparse.py

```diff
@@ -218,14 +241,68 @@ def parse_filter_arg(args, file=sys.stderr):
         if _is_json(args[0]):
             return _parse_json(args[0])
         else:
-            return _with_message(_parse_simple(args[0]), file)
+            key, value = _parse_single(args[0])
+            return _with_message({key: value}, file)
     else:
-        q = {}
-        for i in range(0, len(args), 2):
-            key = args[i]
-            if i + 1 < len(args):
-                value = args[i + 1]
-            else:
-                value = None
-            q.update(_parse_simple(key, value))
+        q = dict(parse_simple(args))
+
         return _with_message(q, file)
+
+
+def _add_prefix(prefix, filter):
+    if filter:
+        for key, value in filter.items():
+            if key in ("$and", "$or"):
+                if isinstance(value, list) or isinstance(value, tuple):
+                    yield key, [dict(_add_prefix(prefix, item)) for item in value]
+                else:
+                    raise ValueError(
+                        "The argument to a logical operator must be a sequence (e.g. a list)!"
```

Comment:
```suggestion
                        "The argument to a logical operator must be a list or tuple!"
```
Since we are not following the normal [EAFP](https://docs.python.org/3.5/glossary.html#term-eafp) pattern here, we know what types we are looking for. Unless `isinstance` will accept like-like and tuple-like objects.

Reply:
This is true. Maybe we should relax this... but I don't want to put in the effort to change this now, so I'm going to accept your suggestion and maybe we can improve this at a later point.